### PR TITLE
HPCC-25164 Unable to spray JSON UTF-8 file that contains BOM

### PR DESCRIPTION
**Pull request by AttilaVamos**
Handle BOM of JSON source file in Spray operation.

Add new Regression Suite case to test spray JSON and XML file with BOM.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested locally.

### CODE REVIEW

**Review by afishbeck — DISMISSED** [review]
@AttilaVamos looks good.

**Comment by richardkchapman** [comment]
Giving a build error:

/usr/bin/ld: CMakeFiles/dalift.dir/fttransform.cpp.o: in function `TransferServer::pull()':
495
/home/runner/work/HPCC-Platform/HPCC-Platform/src/dali/ft/fttransform.cpp:830: undefined reference to `RemoteFilename::queryFileSystemProperties() const'
496
/usr/bin/ld: /home/runner/work/HPCC-Platform/HPCC-Platform/src/dali/ft/fttransform.cpp:902: undefined reference to `RemoteFilename::queryFileSystemProperties() const'
497
collect2: error: ld returned 1 exit status
498

**Review by afishbeck — APPROVED** [review]
@AttilaVamos looks good.

File: dali/ft/filecopy.cpp [hunk 1]
```diff
@@ -1736,8 +1736,12 @@ void FileSprayer::locateJsonHeader(IFileIO * io, unsigned headerSize, offset_t &
     Owned<IFileIOStream> in = createIOStream(io);
 
     JsonSplitter jsplitter(srcFormat, *in);
+    if (headerSize > 0)
+        in->seek(headerSize, IFSbegin); // Skip BOM
     headerLength = jsplitter.getHeaderLength();
     footerLength = jsplitter.getFooterLength();
```
Comment (ghalliday) [inline dali/ft/filecopy.cpp:1742]: not new, but doesn't this read the entire json file?  Is there a more efficient way to calculate this.

File: dali/ft/filecopy.cpp [hunk 2]
```diff
@@ -1736,8 +1736,12 @@ void FileSprayer::locateJsonHeader(IFileIO * io, unsigned headerSize, offset_t &
...
+    if (headerSize > 0)
```
Comment (ghalliday) [inline dali/ft/filecopy.cpp:1743]: This should have a comment to explain why the footer is adjusted with the headerSize.  It looks like it is depending on an implementation detail.  I suspect headerSize should be passed to JsonSplitter instead and the adjustments made within that class.

File: dali/ft/daftformat.ipp [hunk 3]
```diff
@@ -558,10 +558,11 @@ protected:
         JSON_DBGLOG("CJsonInputPartitioner::findSplitPoint: splitOffset %lld", splitOffset);
         JSON_DBGLOG("CJsonInputPartitioner::findSplitPoint: cursor(inputOffset: %lld, nextInputOffset: %lld, outputOffset: %lld, trimLength: %lld",
                         cursor.inputOffset, cursor.nextInputOffset, cursor.outputOffset, cursor.trimLength);
-        cursor.inputOffset = 0;
         if (!splitOffset) //header + 0 is first offset
             return;
 
+        cursor.inputOffset = 0;
```
Comment (ghalliday) [inline dali/ft/daftformat.ipp:564]: Why was this moved?  When spraying an empty json file with a bom marker?

**Review by ghalliday — COMMENTED** [review]
A few minor questions.

**Comment by ghalliday** [comment]
@AttilaVamos waiting for a response...
